src/node/channels.h

```diff
@@ -46,9 +52,57 @@ namespace ccf
   class Channel
   {
   private:
-    // Used for key exchange
-    tls::KeyExchangeContext ctx;
-    ChannelStatus status = INITIATED;
+    class OutgoingQueue
+    {
+    public:
+      struct Msg
+      {
+        NodeMsgType type;
+        std::vector<uint8_t> raw_plain; // To be intergrity-protected
+        std::vector<uint8_t> raw_cipher; // To be encrypted
+
+        Msg(NodeMsgType msg_type, CBuffer raw_plain_, CBuffer raw_cipher_) :
+          type(msg_type),
+          raw_plain(raw_plain_),
+          raw_cipher(raw_cipher_)
+        {}
+      };
+
+    private:
+      static constexpr auto max_queued_msgs = 16;
+      std::queue<Msg> outgoing_msgs;
+      Channel& parent;
+
+    public:
+      OutgoingQueue(Channel& parent_) : parent(parent_) {}
+
+      void queue_msg(Msg&& msg)
+      {
+        if (outgoing_msgs.size() < max_queued_msgs)
+        {
+          outgoing_msgs.emplace(std::move(msg));
+        }
+        else
+        {
+          LOG_DEBUG_FMT(
+            "Cannot queue msg for node as channel is full (max size: {})",
```

Comment:
Why do we have a max?

Reply:
Can we use the same MSPCQ we use elsewhere?

Reply:
I think we don't need the MPSCQ here, since the work should already have been queued onto a single thread?

The max looks like a sensible safeguard. In normal operation we'll only have a handful of queued messages, and if we have more its because establishing a channel is taking a long time (longer than multiple raft heartbeats). In the latter case we're just matching the old behaviour - saying these messages were lost in flight and dropping them - rather than queuing indefinitely.

Reply:
As discussed offline, it's actually OK to cap the size of the queue to 1 (there's no actual need for a queue in that case). There's only one message we don't want to lose (the `RequestVote` when a node becomes candidate) and other messages are resent shortly (e.g. `AppendEntries`).

Reply:
Done now.